app/scripts/metamask-controller.js

```diff
@@ -1934,6 +1934,21 @@ export default class MetamaskController extends EventEmitter {
     cb(null, this.getState());
   }
 
+  /**
+   * Method to return a boolean if the keyring for the currently selected
+   * account is a ledger or trezor keyring.
+   * TODO: remove this method when Ledger and Trezor release their supported
+   * client utilities for EIP-1559
+   * @returns {boolean} true if the keyring type supports EIP-1559
+   */
+  getCurrentAccountEIP1559Compatibility() {
+    const selectedAddress = this.preferencesController.getSelectedAddress();
+    const keyring = this.keyringController.getKeyringForAccount(
+      selectedAddress,
+    );
+    return keyring.type !== 'ledger' && keyring.type !== 'trezor';
```

Comment:
Didn't test this locally so not sure, but should these strings be 'Ledger Hardware' and 'Trezor Hardware'?

Reply:
oh snap, yeah you're right! I was conflating deviceName with keyring type.

Reply:
Good catch @adonesky1 !